src/app/pages/RewardPage/hooks/useGetNextPositiveCheckpoint.ts

```diff
@@ -0,0 +1,62 @@
+import { useAccount } from 'app/hooks/useAccount';
+import { useCallback, useEffect, useState } from 'react';
+import { contractReader } from 'utils/sovryn/contract-reader';
+
+const MAX_NEXT_POSITIVE_CHECKPOINT = 75;
+
+type UserCheckpoint = {
+  checkpointNum: string;
+  hasFees: boolean;
+  hasSkippedCheckpoints: boolean;
+};
+
+export const useGetNextPositiveCheckpoint = (
+  contractAddress: string,
+  totalTokenCheckpoints: number,
+) => {
+  const address = useAccount();
+  const [userCheckpoint, setUserCheckpoint] = useState<UserCheckpoint>();
+
+  const updateNextPositiveCheckpoint = useCallback(async () => {
+    let index = 1;
+
+    while (index < 5) {
```

Comment:
Why 5?

Reply:
I don't think that's true either because he only says that `withdraw` / `withdrawRBTC` methods can process up to 300 checkpoints in one tx. Verify it with @tjcloa pls.

Reply:
max params for getNextPositiveCheckpoint and for withdraw are independent  
`getNextPositiveCheckpoint`: 75 is empirical max value before a node makes the func call fail, so it can digest 75 checkpoints at most per call. we tested it on a local forked mainnet, in theory can be a different number for the actual mainnet. therefore to find the first "claimable" checkpoint with hasFees === true, you need to cycle it in chunks of 75 till you find a claimable checkpoint or process all the checkpoints.
`withdraw functions`: 300 checkpoints at max per call on a local forked mainnet which should be the same as on the mainnet, but could be different in the wild - this is applied when you do actual claiming (withdraw)